cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -446,6 +461,7 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
             var is_proctored_exam;
             var time_limit = this.getExamTimeLimit();
             var exam_review_rules = this.$('#id_exam_review_rules').val();
+            var hide_after_due = this.$('#id_hide_exam_after_due_date').is(':checked');
```

Comment:
Yup, this is intentional. This makes it so the value is persisted, so if you change the exam type away from timed, then back, the value you had previously will remain.